backend/commons/src/main/java/org/sonarsource/sonarlint/core/commons/log/LogOutput.java

```diff
@@ -21,13 +21,29 @@
 
 import java.io.PrintWriter;
 import java.io.StringWriter;
+import javax.annotation.Nullable;
 
 /**
  * Allow to redirect SonarLint logs to a custom output on client side
  */
 public interface LogOutput {
 
-  void log(String formattedMessage, Level level);
+  /**
+   * @deprecated please implement {@link #log(String, Level, String)} instead
+   */
+  @Deprecated(since = "10.0")
+  default void log(String formattedMessage, Level level) {
```

Comment:
Why deprecate this method? I assume most places logging won't have a stack trace, so it could act as a shortcut for the other one

Reply:
The goal is for clients to have only one method to implement. In fact, I could have removed it totally since we are not really concerned about backward compatibility in 10.x...